tests/utils.go

```diff
@@ -1036,10 +1036,10 @@ func NewConsoleExpecter(virtCli kubecli.KubevirtClient, vm *v1.VirtualMachine, t
 	expecterReader, expecterWriter := io.Pipe()
 	resCh := make(chan error)
 	stopChan := make(chan struct{})
-	go func() {
+	go func(vm *v1.VirtualMachine, vmReader *io.PipeReader, expecterWriter *io.PipeWriter, resCh chan error) {
 		err := virtCli.VM(vm.ObjectMeta.Namespace).SerialConsole(vm.ObjectMeta.Name, vmReader, expecterWriter)
 		resCh <- err
-	}()
+	}(vm, vmReader, expecterWriter, resCh)
```

Comment:
Hm, I don't think that this is the issue. All values are written before we run the go-routine and they are not changed afterwards. Do you have a stacktrace which pointed to this go-routine?

Reply:
I'd like to leave this patch even if it doesn't directly fix the problem we were having.

We should get out of the habit of creating goroutines that access variables in the outer scope like that.